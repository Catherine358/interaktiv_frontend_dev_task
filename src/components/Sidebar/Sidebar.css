.sidebar {
    margin-bottom: 20px;
}

@media screen and (max-width: 685px) {
  .sidebar {
    flex: 1;
    padding-left: 10px;
    padding-right: 10px;
  }
}
