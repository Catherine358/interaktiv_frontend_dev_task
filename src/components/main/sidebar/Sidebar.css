.sidebar-container {
    flex: 0.3;
    margin-bottom: 20px;
}

@media screen and (max-width: 530px) {
    .sidebar-container {
        flex: 1;
    }
}
