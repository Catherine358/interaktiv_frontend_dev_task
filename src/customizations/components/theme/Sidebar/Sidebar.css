.sidebar {
    margin-bottom: 20px;
}

@media screen and (max-width: 530px) {
    .sidebar-container {
        flex: 1;
    }
}
