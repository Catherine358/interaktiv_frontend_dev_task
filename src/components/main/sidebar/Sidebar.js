import Navigation from './navigation/Navigation';
import Contact from './contact/Contact';

import './Sidebar.css';

function Sidebar() {
    return (
        <section className="sidebar-container">
            <Navigation />
            <Contact />
        </section>
    );
}

export default Sidebar;
