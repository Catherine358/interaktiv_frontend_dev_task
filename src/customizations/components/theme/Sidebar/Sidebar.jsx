import SidebarNavigation from '../SidebarNavigation/SidebarNavigation';
import ContactNavigation from '../ContactNavigation/ContactNavigation';

import './Sidebar.css';

const Sidebar = () => (
    <section className="sidebar">
      <SidebarNavigation />
      <ContactNavigation />
    </section>
  );

export default Sidebar;
