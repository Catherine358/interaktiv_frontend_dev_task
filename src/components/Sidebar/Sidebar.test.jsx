import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Sidebar from './Sidebar';

describe('Sidebar', () => {
  it('renders sidebar', () => {
    const component = render(
      <Sidebar  />
    );
    expect(component).toMatchSnapshot();
    expect(screen.getByText(/navigation/i)).toBeInTheDocument();
    expect(screen.getByText(/contact/i)).toBeInTheDocument();
  });
});
