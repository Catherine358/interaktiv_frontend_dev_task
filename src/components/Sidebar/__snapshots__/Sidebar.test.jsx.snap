// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders sidebar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sidebar"
      >
        <section
          class="sidebar-nav"
        >
          <div
            class="title-container"
          >
            <p>
              Navigation
            </p>
          </div>
          <div
            class="ui segment"
            data-testid="navigation-item"
            horizontal="true"
          >
            <div
              class="small-card-img-container"
            >
              <img
                alt="Money Laundering and Fraud Prevention"
                class="ui image"
                height="35"
                src="fraud_prevention.png"
                width="35"
              />
            </div>
            <p>
              Money Laundering and Fraud Prevention
            </p>
          </div>
          <div
            class="ui segment"
            data-testid="navigation-item"
            horizontal="true"
          >
            <div
              class="small-card-img-container"
            >
              <img
                alt="IT security and information security"
                class="ui image"
                height="35"
                src="it_security.png"
                width="35"
              />
            </div>
            <p>
              IT security and information security
            </p>
          </div>
          <div
            class="ui segment"
            data-testid="navigation-item"
            horizontal="true"
          >
            <div
              class="small-card-img-container"
            >
              <img
                alt="Data Protection"
                class="ui image"
                height="35"
                src="data_protection.png"
                width="35"
              />
            </div>
            <p>
              Data Protection
            </p>
          </div>
          <div
            class="ui segment"
            data-testid="navigation-item"
            horizontal="true"
          >
            <div
              class="small-card-img-container"
            >
              <img
                alt="WpHG-Compliance"
                class="ui image"
                height="35"
                src="wphg.png"
                width="35"
              />
            </div>
            <p>
              WpHG-Compliance
            </p>
          </div>
        </section>
        <section>
          <div
            class="title-container"
          >
            <p>
              Contact
            </p>
          </div>
          <div
            class="sidebar-contact-el"
          >
            <p
              class="sidebar-contact-el-title"
            >
              Technical support
            </p>
            <p
              class="sidebar-contact-el-text"
            >
              John Doe
            </p>
            <a
              class="sidebar-contact-el-email"
            >
              it@example.com
            </a>
          </div>
          <div
            class="sidebar-contact-el"
          >
            <p
              class="sidebar-contact-el-title"
            >
              Content-didactic topics
            </p>
            <p
              class="sidebar-contact-el-text"
            >
              Jane Roe
            </p>
            <a
              class="sidebar-contact-el-email"
            >
              content@example.com
            </a>
          </div>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sidebar"
    >
      <section
        class="sidebar-nav"
      >
        <div
          class="title-container"
        >
          <p>
            Navigation
          </p>
        </div>
        <div
          class="ui segment"
          data-testid="navigation-item"
          horizontal="true"
        >
          <div
            class="small-card-img-container"
          >
            <img
              alt="Money Laundering and Fraud Prevention"
              class="ui image"
              height="35"
              src="fraud_prevention.png"
              width="35"
            />
          </div>
          <p>
            Money Laundering and Fraud Prevention
          </p>
        </div>
        <div
          class="ui segment"
          data-testid="navigation-item"
          horizontal="true"
        >
          <div
            class="small-card-img-container"
          >
            <img
              alt="IT security and information security"
              class="ui image"
              height="35"
              src="it_security.png"
              width="35"
            />
          </div>
          <p>
            IT security and information security
          </p>
        </div>
        <div
          class="ui segment"
          data-testid="navigation-item"
          horizontal="true"
        >
          <div
            class="small-card-img-container"
          >
            <img
              alt="Data Protection"
              class="ui image"
              height="35"
              src="data_protection.png"
              width="35"
            />
          </div>
          <p>
            Data Protection
          </p>
        </div>
        <div
          class="ui segment"
          data-testid="navigation-item"
          horizontal="true"
        >
          <div
            class="small-card-img-container"
          >
            <img
              alt="WpHG-Compliance"
              class="ui image"
              height="35"
              src="wphg.png"
              width="35"
            />
          </div>
          <p>
            WpHG-Compliance
          </p>
        </div>
      </section>
      <section>
        <div
          class="title-container"
        >
          <p>
            Contact
          </p>
        </div>
        <div
          class="sidebar-contact-el"
        >
          <p
            class="sidebar-contact-el-title"
          >
            Technical support
          </p>
          <p
            class="sidebar-contact-el-text"
          >
            John Doe
          </p>
          <a
            class="sidebar-contact-el-email"
          >
            it@example.com
          </a>
        </div>
        <div
          class="sidebar-contact-el"
        >
          <p
            class="sidebar-contact-el-title"
          >
            Content-didactic topics
          </p>
          <p
            class="sidebar-contact-el-text"
          >
            Jane Roe
          </p>
          <a
            class="sidebar-contact-el-email"
          >
            content@example.com
          </a>
        </div>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
